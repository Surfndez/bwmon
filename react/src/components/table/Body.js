import React from 'react';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';

const Body = ({ prefix, cellInfos, values }) => (
    <TableBody>
        {values.map((value, index) => (
            <TableRow
                key={value.id}
                id={prefix + '-data-' + index}
                style={index % 2 ? { background: 'ghostwhite' } : { background: 'white' }}
            >
                {cellInfos.map((cellInfo) => {
                    return (
                        <TableCell key={cellInfo.id} align={cellInfo.align}>
                            {cellInfo.convert ? cellInfo.convert(value[cellInfo.id]) : value[cellInfo.id]}
                        </TableCell>
                    );
                })}
            </TableRow>
        ))}
    </TableBody>
);

export default Body;
