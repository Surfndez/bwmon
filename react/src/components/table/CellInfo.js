export default class CellInfo {
    constructor(id, sortable, align, label, footer, convert) {
        this.id = id;
        this.sortable = sortable;
        this.align = align;
        this.label = label;
        this.footer = footer;
        this.convert = convert;
    }
}
