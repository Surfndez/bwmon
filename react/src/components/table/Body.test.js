import React from 'react';
import { create } from 'react-test-renderer';
import Body from './Body';
import CellInfo from './CellInfo';

describe('Body', () => {
    const convert = (value) => value / 1000;

    const cellInfos = [new CellInfo('IP', true, 'left', 'IP', false), new CellInfo('up', false, 'right', 'Down', true, convert)];

    const values = [
        { id: 1, IP: '192.168.1.14', up: 12345202809 },
        { id: 2, IP: '192.168.1.21', up: 123453907 },
        { id: 3, IP: '192.168.1.10', up: 5370488 },
    ];

    test('renders correctly', () => {
        const tree = create(<Body prefix="test" cellInfos={cellInfos} values={values} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
