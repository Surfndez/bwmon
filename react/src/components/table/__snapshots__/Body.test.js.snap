// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders correctly 1`] = `
<tbody
  className="MuiTableBody-root"
  role={null}
>
  <tr
    className="MuiTableRow-root"
    id="test-data-0"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      192.168.1.14
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
    >
      12345202.809
    </td>
  </tr>
  <tr
    className="MuiTableRow-root"
    id="test-data-1"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      192.168.1.21
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
    >
      123453.907
    </td>
  </tr>
  <tr
    className="MuiTableRow-root"
    id="test-data-2"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      192.168.1.10
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
    >
      5370.488
    </td>
  </tr>
</tbody>
`;
